jenkins_jobs/modules/parameters.py

```diff
@@ -1143,6 +1143,229 @@ def param_separator(registry, xml_parent, data):
+
+        :Parameter: * **groovy** (`str`) Groovy DSL Script
+                    * **use-groovy-sandbox** (`bool`) To run this
+                        Groovy script in a sandbox with limited abilities
+                    * **script-additional-classpath** (`list`) Additional
+                        classpath entries accessible from the script.
+    :arg list fallback-script: Use a Fallback script. If the script
+        (specified above) fails, the fallback script will be used as a fallback.
+
+        :Parameter: * **groovy** (`str`) Groovy DSL Script
+                    * **use-groovy-sandbox** (`bool`) To run this Groovy
+                        script in a sandbox with limited abilities.
+                    * **script-additional-classpath** (`list`) Additional
+                        classpath entries accessible from the script.
+    :arg bool enable-filters: If enabled a text box will appear next to
+        this element and will permit the user to filter its entries. The
+        list values never get re-evaluated.
+    :arg int filter-starts-at: How many characters a user must enter
+        before the filter is applied.
+    :arg str choice-type: type of the choices. (default 'PT_SINGLE_SELECT')
+
+        :Allowed Values: * **PT_SINGLE_SELECT**
+                    * **PT_MULTI_SELECT**
+                    * **PT_RADIO**
+                    * **PT_CHECKBOX**
+    :arg str project-name: the name of job from which the user can pick runs
+    :arg str full-project-name: the full project name of job from which the
+        user can pick runs
+
+    Example::
+
+      parameters:
+        - active-choices:
+            name: lorem
+            description: ipsum
+            script:
+                groovy: |-
+                    return [
+                        'param1',
+                        'param2'
+                    ]
+                use-groovy-sandbox: 'true'
+                script-additional-classpath:
+                    - file:/jar-file-path
+                    - file:/jar-file-path2
+            fallback-script:
+                groovy: |-
+                    return [
+                        'param3',
+                        'param4'
+                    ]
+                use-groovy-sandbox: 'true'
+                script-additional-classpath:
+                    - file:/jar-file-path
+                    - file:/jar-file-path2
+            choice-type: PT_MULTI_SELECT
+            enable-filters: False
+            filter-starts-at: 1
+            project-name: job-name-test
+            project-full-name: my_jobs/job_folder1/job-name-test
+    """
+    element_name = "org.biouno.unochoice.ChoiceParameter"
+    pdef = XML.SubElement(xml_parent, element_name)
+
+    valid_choice_types = [
+        "PT_SINGLE_SELECT",
+        "PT_MULTI_SELECT",
+        "PT_RADIO",
+        "PT_CHECKBOX",
+    ]
+
+    mapping = [
+        ("name", "name", ""),
```

Comment:
I believe a parameter must have a name so the default value should not be specified - otherwise how do I access the value of this parameter in my script?